### DISCUSSION TITLE
Slow transcribing on g4dn instances

### DISCUSSION THREAD

**Discussion by ahomatthew** [openai/whisper#939] [Q&A] [answered]
I have a simple test running in a Kubernetes pod running on a g4dn.xlarge node which has 1 NVIDIA T4 GPUs and 4 custom Intel Cascade Lake vCPUs -- it's the only service running on the node aside from the log forwarder, and i'm only achieving 2x audio speed for `small.en`.  I.e. I have a 50 second audio clip taking 25 seconds to finish transcribing. 

On my local machine I was getting 20x real-time audio speed.  Has anyone tried using this before in a Kubernetes cluster? Is there some way to speed this up like a configuration I need?  If i had to guess it's just not using the GPU? 

Any help would be appreciated.

**Comment by ahomatthew** [comment, marked as answer]
Needed to install the CUDA drivers in the container.

**Reply by ghost** [reply]
What is the speed you are able to achieve now? Considering it was 20x on your local machine, I would hope its significantly higher on the T4 GPU

**Reply by ahomatthew** [reply]
Well, I ran it on a 2022 M1 Max with a 32-core gpu, so it's pretty nice for a local.  I think on the g4dn.xlarge I was testing on I was seeing very similar times, in the 16-24x range.

**Reply by ryanheise** [reply]
I believe Google Colab has T4 GPUs so you could also try running some speed tests there.
